src/daemon/ca/manager.rs

```diff
@@ -831,10 +831,48 @@ impl CaManager {
             .filter(|secs| started < Timestamp::now_minus_seconds(*secs));
 
         if let Ok(ca) = self.get_ca(&ca_handle).await {
-            for parent in ca.parents() {
-                updates.push(self.ca_sync_parent_infallible(ca_handle.clone(), parent.clone(), actor.clone()));
+            // get updates from parents
+            {
+                if ca.nr_parents() <= self.config.ca_refresh_parents_batch_size {
+                    // Nr of parents is below batch size, so just process all of them
+                    for parent in ca.parents() {
+                        updates.push(self.ca_sync_parent_infallible(ca_handle.clone(), parent.clone(), actor.clone()));
+                    }
+                } else {
+                    // more parents than the batch size exist, so sort them:
+                    // - take the ones with no last exchanges first
+                    // - then sort by last exchange, minute grade granularity - oldest first
+                    //    - where failures come before success within the same minute
+                    // - then take the first N parents for this batch
+                    match self.status_store.lock().await.get_ca_status(&ca_handle).await {
+                        Err(e) => {
+                            panic!("System level error encountered while updating ca status: {}", e);
+                        }
+                        Ok(status) => {
```

Comment:
You don't seem to check the status here. Perhaps I misunderstand but it looks like you'll take some of the parents and sync them then exit this function, so what happens to the remaining parents? If you come back in here again they will have an older refresh time than the ones that were just refreshed, but they will still all be in the total set and so all synced Is there some logic somewhere else that makes that a NO-OP or do you need to check the status returned by `get_ca_status()` here?

Reply:
I use the `status` to get the current parents for which a status is known and sort them:
```
let mut parents_by_last_exchange = status.parents().sorted_by_last_exchange();
```

The remaining parents will be first next time.. this is also why I don't sort by last success only, but last exchange.. otherwise continuously failing parents could starve the others.

I am adding the parents for which there is no recorded status yet, i.e. new parents, first so that they take precedence. Since parent status is only created *after* the first sync attempt.

Reply:
But will not the already handled parents also be included next time?

Reply:
By example: Default batch size is 25 parents. If you have 27 then you will do 25 first, and leave 2, and then in t he next run (10 minutes by default) do 25 again: the 2 remaining and 23 that were also processed in the previous batch.

Reply:
Yes. So is right that the additional 23 are also processed?

Reply:
yes, sorry.. missing info might be that it is always safe to sync with a parent. This is a no-op in case the entitlements have not changed. Along the same lines.. if you have 1 parent it's perfectly fine to sync it in every run.

Some might say that the default of 10 mins is aggressive, but I have not heard of any issues from parents.

If a CA has 100s of parents, and they use defaults then it will take longer for the parents to be resync-ed. E.g. with 100 parents -> every 40 minutes. But this is not really an issue either. You kind of want to stay below the hour I think - but there are no formal RFC requirements here - every hour would be good because you want to learn about changes in resources soon-ish.

In the unlikely case that a CA has that many parents, they can also tweak the defaults - e.g. refresh 25 parents every 5 minutes instead of every 10 minutes.

Reply:
Thanks.